**Titanic Survivor Analysis**

In [43]:
#importing librariess 
import numpy as np 
import pandas as pd 

Reading the dataset

In [44]:
data = pd.DataFrame(pd.read_csv('/content/train.csv'))

In [45]:
data.head() # get first 5 rows 

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

In [46]:
data.tail() # get last 5 rows

PassengerId  Survived  Pclass  ...   Fare Cabin  Embarked
886          887         0       2  ...  13.00   NaN         S
887          888         1       1  ...  30.00   B42         S
888          889         0       3  ...  23.45   NaN         S
889          890         1       1  ...  30.00  C148         C
890          891         0       3  ...   7.75   NaN         Q

[5 rows x 12 columns]

In [48]:
#dimension of the dataset 
data.shape

(891, 12)

In [49]:
# statistical measurement of the dataset 
data.describe()

PassengerId    Survived      Pclass  ...       SibSp       Parch        Fare
count   891.000000  891.000000  891.000000  ...  891.000000  891.000000  891.000000
mean    446.000000    0.383838    2.308642  ...    0.523008    0.381594   32.204208
std     257.353842    0.486592    0.836071  ...    1.102743    0.806057   49.693429
min       1.000000    0.000000    1.000000  ...    0.000000    0.000000    0.000000
25%     223.500000    0.000000    2.000000  ...    0.000000    0.000000    7.910400
50%     446.000000    0.000000    3.000000  ...    0.000000    0.000000   14.454200
75%     668.500000    1.000000    3.000000  ...    1.000000    0.000000   31.000000
max     891.000000    1.000000    3.000000  ...    8.000000    6.000000  512.329200

[8 rows x 7 columns]

In [50]:
# check concise summary(like datatype,column type and non null values) of dataframe 
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [51]:
# check for null value of dataframe 
print(data.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


Now we can see that around 20% of the data is missing in Age column and in Cabin column there is 77% data is missing . That is huge in number . Usually  we can't eliminate those values cause that creates a huge error in my Dataset.

Now we seperate those columns which have more than 35% of the values are missing in the Dataset

In [52]:
x= data.isnull().sum()
drop_col=x[x>(35/100 * data.shape[0])]
drop_col

Cabin    687
dtype: int64

Here we can see that Cabin column is providing just the location of the ship which is string / object  in datatype and here is no integer datatype associated in the Cabin Column . So for this reason we can eliminate this Cabin Column cause it has no use and it doesn't affect the analysis of the dataset

In [53]:
drop_col.index

Index(['Cabin'], dtype='object')

In [55]:
data.drop(drop_col.index,axis = 1,inplace =True)
data.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Embarked         2
dtype: int64

Here we can see that Age has multiple null values .But here we can't remove Age column as it carries float datatype. If we remove this column that creats a huge error in my dataset. that's why we are replacing those null values by the median values of the respective column. Here i am using median. We know that median provides better result over mean as median is the middle most value of the incresing or decresing order of values.

In [59]:
data.median()

PassengerId    446.0000
Survived         0.0000
Pclass           3.0000
Age             28.0000
SibSp            0.0000
Parch            0.0000
Fare            14.4542
dtype: float64

In [57]:
data.fillna(data.median(), inplace= True)
data.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       2
dtype: int64

Here we can remove Embarked column as it contains string values and we can't calculate median of string values. But we can't remove this column cause based on that we have to reach on the conclusion about the dataset.

In [60]:
data["Embarked"].describe() 

count     889
unique      3
top         S
freq      644
Name: Embarked, dtype: object

Now we will replace the null values of Embarked column by the modal string / value . Here we can see in the Embarked coloumn that mode is "S" (which has a highest frequency)

In [62]:
data['Embarked'].fillna("S", inplace=True)
data.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

Now we can see that there is no null values in  my dataframe. So no data Cleaning is required.

In [64]:
data.corr() # Correlation of the attributes 

PassengerId  Survived    Pclass  ...     SibSp     Parch      Fare
PassengerId     1.000000 -0.005007 -0.035144  ... -0.057527 -0.001652  0.012658
Survived       -0.005007  1.000000 -0.338481  ... -0.035322  0.081629  0.257307
Pclass         -0.035144 -0.338481  1.000000  ...  0.083081  0.018443 -0.549500
Age             0.034212 -0.064910 -0.339898  ... -0.233296 -0.172482  0.096688
SibSp          -0.057527 -0.035322  0.083081  ...  1.000000  0.414838  0.159651
Parch          -0.001652  0.081629  0.018443  ...  0.414838  1.000000  0.216225
Fare            0.012658  0.257307 -0.549500  ...  0.159651  0.216225  1.000000

[7 rows x 7 columns]

Now , Here we can see that in between SibSp : 'Number of Siblings / Spouse  aboard' and Parch : 'Number of parents / Children abroad' have same amount of Correlation. Which signify that both are same column. Now we replacing these two column by adding these two columns and will get a New Column.  

In [65]:
data['familysize'] = data['SibSp'] + data['Parch']
data.drop(['SibSp','Parch'],axis=1,inplace=True)
data.corr()

PassengerId  Survived    Pclass       Age      Fare  familysize
PassengerId     1.000000 -0.005007 -0.035144  0.034212  0.012658   -0.040143
Survived       -0.005007  1.000000 -0.338481 -0.064910  0.257307    0.016639
Pclass         -0.035144 -0.338481  1.000000 -0.339898 -0.549500    0.065997
Age             0.034212 -0.064910 -0.339898  1.000000  0.096688   -0.245619
Fare            0.012658  0.257307 -0.549500  0.096688  1.000000    0.217138
familysize     -0.040143  0.016639  0.065997 -0.245619  0.217138    1.000000

familysize in the ship doesn't have much correlation with survival rate .

Now we want to check whether the person was alone or not can affect the survival rate .

In [67]:
data['alone']=[0 if data['familysize'][i]>0 else 1 for i in data.index] # 0 if not alone and 1 if alone

In [68]:
data.head()

PassengerId  Survived  Pclass  ... Embarked familysize  alone
0            1         0       3  ...        S          1      0
1            2         1       1  ...        C          1      0
2            3         1       3  ...        S          0      1
3            4         1       1  ...        S          1      0
4            5         0       3  ...        S          0      1

[5 rows x 11 columns]

In [69]:
data.tail()

PassengerId  Survived  Pclass  ... Embarked familysize  alone
886          887         0       2  ...        S          0      1
887          888         1       1  ...        S          0      1
888          889         0       3  ...        S          3      0
889          890         1       1  ...        C          0      1
890          891         0       3  ...        Q          0      1

[5 rows x 11 columns]

In [78]:
data.groupby(['alone'])["Survived"].mean()

alone
0    0.505650
1    0.303538
Name: Survived, dtype: float64

In [79]:
data.groupby(['alone'])["Pclass"].mean()

alone
0    2.169492
1    2.400372
Name: Pclass, dtype: float64

In [80]:
data.groupby(['alone'])['Fare'].mean()

alone
0    48.832275
1    21.242689
Name: Fare, dtype: float64

If the person is alone has a lesser chance of surviving than those persons who are not alone. Cause those who are not alone they have a family , so its much likely that they will stay in better Passenger Class and also which leads to much prioritise over others as they have a child(whom survival is much neccesary)

In [74]:
data[['alone','Fare']].corr()

alone      Fare
alone  1.000000 -0.271832
Fare  -0.271832  1.000000

We can comment that if the person is alone then ticket price is lower 

In [84]:
data.groupby(['Pclass'])["Survived"].mean()

Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64

Passenger with higher class (Survival Rate = 63%) > Middle Class(Survival Rate =47%) > Lower Class (Survival Rate = 24%)

In [81]:
data['sex']=[0 if data['Sex'][i]=='male'else 1 for i in data.index] # 1 for female and 0 for male

In [82]:
data.groupby(['sex'])['Survived'].mean()

sex
0    0.188908
1    0.742038
Name: Survived, dtype: float64

We can comment that very large number of female has survived than male . 
So Women was prioritized over Men. That signify a good sign 

In [83]:
data.groupby(['Embarked'])['Survived'].mean()

Embarked
C    0.553571
Q    0.389610
S    0.339009
Name: Survived, dtype: float64

Those people who boarded from Cherbourg Port has maximum Survival Rate(55%) than those who stared from Queenstown Port and Southampton Port respectively.



We can Comment that 
Cherbourg Port(Survival Rate = 55%) > Queenstown Port(Survival Rate = 39%) >
Southampton Port(Survival Rate = 34%)

**COMMENT**



1. Those people who are not alone has a higher chance of surviving than those  
    people who are alone.  
2. People who stayed in Higher Class i.e Rich People have Higher Survival Rate 
    than Middle Class and Lower Class respectively.The hierarichymight have 
    followed while saving the passengers.
3. Significant large number of female has survived than male . So Women was 
    prioritized over Men.
4. Those people who boarded from Cherbourg Port has Maximum Survival Rate(55%) 
    than those who stared from Queenstown Port and Southampton Port respectively
     
